incompatible version of matplotlib installed when running a clean install of cellxgene

`scanpy 1.4.4.post1 has requirement matplotlib==3.0.*, but you'll have matplotlib 3.1.1 which is incompatible.`


Full log:

```zsh
> cellxgene@0.11.0 backend-dev /Users/sbadajoz/dev/cellxgene/client
> python3.6 -m venv cellxgene && source cellxgene/bin/activate && yes | pip uninstall cellxgene || true && pip install -e .. && cellxgene launch  "/Users/sbadajoz/Downloads/TM_facs_processed.h5ad"

sh: python3.6: command not found
Obtaining file:///Users/sbadajoz/dev/cellxgene
Collecting anndata>=0.6.20 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/2b/72/87196c15f68d9865c31a43a10cf7c50bcbcedd5607d09f9aada0b3963103/anndata-0.6.22.post1-py3-none-any.whl
Collecting click>=6.7 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl
Collecting Flask>=1.0.2 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/9b/93/628509b8d5dc749656a9641f4caf13540e2cdec85276964ff8f43bbb1d3b/Flask-1.1.1-py2.py3-none-any.whl
Collecting Flask-Caching>=1.4.0 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/bd/4e/c701bdcd566f26187398cf058ad2ca14c8b2f50b17a17f9076794ae2f8db/Flask_Caching-1.7.2-py2.py3-none-any.whl
Collecting Flask-Compress>=1.4.0 (from cellxgene==0.11.0)
Collecting Flask-Cors>=3.0.6 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/78/38/e68b11daa5d613e3a91e4bf3da76c94ac9ee0d9cd515af9c1ab80d36f709/Flask_Cors-3.0.8-py2.py3-none-any.whl
Collecting Flask-RESTful>=0.3.6 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/17/44/6e490150ee443ca81d5f88b61bb4bbb133d44d75b0b716ebe92489508da4/Flask_RESTful-0.3.7-py2.py3-none-any.whl
Collecting flatbuffers>=1.10.0 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/c9/84/adf5837f96c39990bc55afdfddf460b38b4562f50341359afa32e4a98de7/flatbuffers-1.11-py2.py3-none-any.whl
Collecting matplotlib>=2.2 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/c3/8b/af9e0984f5c0df06d3fab0bf396eb09cbf05f8452de4e9502b182f59c33b/matplotlib-3.1.1-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting numpy>=1.15.2 (from cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/c1/4b/78119133136c20e5ad2e01bf72b0633241defd619939908223cd394a9c32/numpy-1.17.0-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (15.0MB)
    100% |████████████████████████████████| 15.0MB 2.5MB/s 
Collecting pandas>=0.23.1 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/39/b7/441375a152f3f9929ff8bc2915218ff1a063a59d7137ae0546db616749f9/pandas-0.25.0-cp37-cp37m-macosx_10_9_x86_64.macosx_10_10_x86_64.whl
Collecting scanpy>=1.3.7 (from cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/c9/df/e0957422a85e924501afbff2f9283e02d93cefcb04aa8319ade7af68d266/scanpy-1.4.4.post1-py3-none-any.whl (1.9MB)
    100% |████████████████████████████████| 1.9MB 9.2MB/s 
Collecting scipy>=1.1.0 (from cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/d5/06/1a696649f4b2e706c509cb9333fdc6331fbe71251cede945f9e1fa13ea34/scipy-1.3.1-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (27.7MB)
    100% |████████████████████████████████| 27.7MB 1.8MB/s 
Collecting scikit-learn!=0.20.0,>=0.19.1 (from cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/e9/57/8a9889d49d0d77905af5a7524fb2b468d2ef5fc723684f51f5ca63efed0d/scikit_learn-0.21.3-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (10.5MB)
    100% |████████████████████████████████| 10.5MB 2.9MB/s 
Collecting tables==3.5.1 (from cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/5a/69/08095fcf71b9e51f5bd77419c101dc11a9e0aeb016ae07a0265cf6bfad09/tables-3.5.1-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting h5py (from anndata>=0.6.20->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/80/c5/eec74d7324628f1b640c6e706981c4ed51afcaa1656ece26cb08d862598e/h5py-2.9.0-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting natsort (from anndata>=0.6.20->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/e7/13/a66bfa0ebf00e17778ca0319d081be686a33384d1f612fc8e0fc542ac5d8/natsort-6.0.0-py2.py3-none-any.whl
Collecting itsdangerous>=0.24 (from Flask>=1.0.2->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/76/ae/44b03b253d6fade317f32c24d100b3b35c2239807046a4c953c7b89fa49e/itsdangerous-1.1.0-py2.py3-none-any.whl
Collecting Werkzeug>=0.15 (from Flask>=1.0.2->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/d1/ab/d3bed6b92042622d24decc7aadc8877badf18aeca1571045840ad4956d3f/Werkzeug-0.15.5-py2.py3-none-any.whl
Collecting Jinja2>=2.10.1 (from Flask>=1.0.2->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/1d/e7/fd8b501e7a6dfe492a433deb7b9d833d39ca74916fa8bc63dd1a4947a671/Jinja2-2.10.1-py2.py3-none-any.whl
Collecting Six (from Flask-Cors>=3.0.6->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting pytz (from Flask-RESTful>=0.3.6->cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/87/76/46d697698a143e05f77bec5a526bf4e56a0be61d63425b68f4ba553b51f2/pytz-2019.2-py2.py3-none-any.whl (508kB)
    100% |████████████████████████████████| 512kB 18.9MB/s 
Collecting aniso8601>=0.82 (from Flask-RESTful>=0.3.6->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/45/a4/b4fcadbdab46c2ec2d2f6f8b4ab3f64fd0040789ac7f065eba82119cd602/aniso8601-7.0.0-py2.py3-none-any.whl
Collecting kiwisolver>=1.0.1 (from matplotlib>=2.2->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/df/93/8bc9b52a8846be2b9572aa0a7c881930939b06e4abe1162da6a0430b794f/kiwisolver-1.1.0-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 (from matplotlib>=2.2->cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/11/fa/0160cd525c62d7abd076a070ff02b2b94de589f1a9789774f17d7c54058e/pyparsing-2.4.2-py2.py3-none-any.whl (65kB)
    100% |████████████████████████████████| 71kB 20.5MB/s 
Collecting cycler>=0.10 (from matplotlib>=2.2->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/f7/d2/e07d3ebb2bd7af696440ce7e754c59dd546ffe1bbe732c8ab68b9c834e61/cycler-0.10.0-py2.py3-none-any.whl
Collecting python-dateutil>=2.1 (from matplotlib>=2.2->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/41/17/c62faccbfbd163c7f57f3844689e3a78bae1f403648a6afb1d0866d87fbb/python_dateutil-2.8.0-py2.py3-none-any.whl
Collecting numba>=0.41.0 (from scanpy>=1.3.7->cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/bb/0d/db1d84ec79b223dedb72d7f1823a77797494348fea4b1809d92affea720a/numba-0.45.1-cp37-cp37m-macosx_10_9_x86_64.whl (1.9MB)
    100% |████████████████████████████████| 1.9MB 7.5MB/s 
Collecting tqdm (from scanpy>=1.3.7->cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/02/56/60a5b1c2e634d8e4ff89c7bab47645604e19658f448050a21facffd43796/tqdm-4.33.0-py2.py3-none-any.whl (50kB)
    100% |████████████████████████████████| 51kB 14.0MB/s 
Collecting networkx (from scanpy>=1.3.7->cellxgene==0.11.0)
Collecting seaborn (from scanpy>=1.3.7->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/a8/76/220ba4420459d9c4c9c9587c6ce607bf56c25b3d3d2de62056efe482dadc/seaborn-0.9.0-py3-none-any.whl
Collecting joblib (from scanpy>=1.3.7->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/cd/c1/50a758e8247561e58cb87305b1e90b171b8c767b15b12a1734001f41d356/joblib-0.13.2-py2.py3-none-any.whl
Collecting umap-learn>=0.3.0 (from scanpy>=1.3.7->cellxgene==0.11.0)
Collecting importlib-metadata>=0.7; python_version < "3.8" (from scanpy>=1.3.7->cellxgene==0.11.0)
  Downloading https://files.pythonhosted.org/packages/ad/aa/25fcbded2ab4ed4ff3071d1e000cd4f8f9c65653d2d7157dd105a8e81d42/importlib_metadata-0.19-py2.py3-none-any.whl
Collecting statsmodels>=0.10.0rc2 (from scanpy>=1.3.7->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/b5/b8/50f9b86bbd87b1de961f439c2b93dfc41dd0cb9d65f6b7d824b287b50b21/statsmodels-0.10.1-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting patsy (from scanpy>=1.3.7->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/ea/0c/5f61f1a3d4385d6bf83b83ea495068857ff8dfb89e74824c6e9eb63286d8/patsy-0.5.1-py2.py3-none-any.whl
Collecting mock>=2.0 (from tables==3.5.1->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/05/d2/f94e68be6b17f46d2c353564da56e6fb89ef09faeeff3313a046cb810ca9/mock-3.0.5-py2.py3-none-any.whl
Collecting numexpr>=2.6.2 (from tables==3.5.1->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/71/3b/81472cf68cb8fa28cfd1f6ea2b55d912201f344afbcb9672437ff4966c12/numexpr-2.6.9-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting MarkupSafe>=0.23 (from Jinja2>=2.10.1->Flask>=1.0.2->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/ce/c6/f000f1af136ef74e4a95e33785921c73595c5390403f102e9b231b065b7a/MarkupSafe-1.1.1-cp37-cp37m-macosx_10_6_intel.whl
Requirement already satisfied: setuptools in /Users/sbadajoz/venv/lib/python3.7/site-packages (from kiwisolver>=1.0.1->matplotlib>=2.2->cellxgene==0.11.0) (40.8.0)
Collecting llvmlite>=0.29.0dev0 (from numba>=0.41.0->scanpy>=1.3.7->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/d7/45/9216cdbf71b94ae8eefe24cfbdf4c1b9045a58f297c6e2eab5cb8d05faf3/llvmlite-0.29.0-cp37-cp37m-macosx_10_9_x86_64.whl
Collecting decorator>=4.3.0 (from networkx->scanpy>=1.3.7->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/5f/88/0075e461560a1e750a0dcbf77f1d9de775028c37a19a346a6c565a257399/decorator-4.4.0-py2.py3-none-any.whl
Collecting zipp>=0.5 (from importlib-metadata>=0.7; python_version < "3.8"->scanpy>=1.3.7->cellxgene==0.11.0)
  Using cached https://files.pythonhosted.org/packages/da/bd/1a5fdf15aa44231fd09f63ecf175b60f057ae37ec65b343bb009364923f3/zipp-0.5.2-py2.py3-none-any.whl
scanpy 1.4.4.post1 has requirement matplotlib==3.0.*, but you'll have matplotlib 3.1.1 which is incompatible.
Installing collected packages: pytz, numpy, Six, python-dateutil, pandas, h5py, natsort, scipy, anndata, click, itsdangerous, Werkzeug, MarkupSafe, Jinja2, Flask, Flask-Caching, Flask-Compress, Flask-Cors, aniso8601, Flask-RESTful, flatbuffers, kiwisolver, pyparsing, cycler, matplotlib, mock, numexpr, tables, llvmlite, numba, tqdm, decorator, networkx, joblib, scikit-learn, seaborn, umap-learn, zipp, importlib-metadata, patsy, statsmodels, scanpy, cellxgene
  Running setup.py develop for cellxgene
Successfully installed Flask-1.1.1 Flask-Caching-1.7.2 Flask-Compress-1.4.0 Flask-Cors-3.0.8 Flask-RESTful-0.3.7 Jinja2-2.10.1 MarkupSafe-1.1.1 Six-1.12.0 Werkzeug-0.15.5 aniso8601-7.0.0 anndata-0.6.22.post1 cellxgene click-7.0 cycler-0.10.0 decorator-4.4.0 flatbuffers-1.11 h5py-2.9.0 importlib-metadata-0.19 itsdangerous-1.1.0 joblib-0.13.2 kiwisolver-1.1.0 llvmlite-0.29.0 matplotlib-3.1.1 mock-3.0.5 natsort-6.0.0 networkx-2.3 numba-0.45.1 numexpr-2.6.9 numpy-1.17.0 pandas-0.25.0 patsy-0.5.1 pyparsing-2.4.2 python-dateutil-2.8.0 pytz-2019.2 scanpy-1.4.4.post1 scikit-learn-0.21.3 scipy-1.3.1 seaborn-0.9.0 statsmodels-0.10.1 tables-3.5.1 tqdm-4.33.0 umap-learn-0.3.9 zipp-0.5.2
You are using pip version 19.0.3, however version 19.2.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Traceback (most recent call last):
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (matplotlib 3.1.1 (/Users/sbadajoz/venv/lib/python3.7/site-packages), Requirement.parse('matplotlib==3.0.*'), {'scanpy'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sbadajoz/venv/bin/cellxgene", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3191, in <module>
    @_call_aside
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3175, in _call_aside
    f(*args, **kwargs)
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3204, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 585, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 598, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/sbadajoz/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (matplotlib 3.1.1 (/Users/sbadajoz/venv/lib/python3.7/site-packages), Requirement.parse('matplotlib==3.0.*'), {'scanpy'})
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! cellxgene@0.11.0 backend-dev: `python3.6 -m venv cellxgene && source cellxgene/bin/activate && yes | pip uninstall cellxgene || true && pip install -e .. && cellxgene launch  "/Users/sbadajoz/Downloads/TM_facs_processed.h5ad"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the cellxgene@0.11.0 backend-dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/sbadajoz/.npm/_logs/2019-08-12T17_01_36_406Z-debug.log
```

--- a/server/requirements.txt
+++ b/server/requirements.txt
@@ -6,10 +6,10 @@ Flask-Compress>=1.4.0
 Flask-Cors>=3.0.6
 Flask-RESTful>=0.3.6
 flatbuffers>=1.10.0
+scanpy>=1.3.7
 matplotlib>=2.2
 numpy>=1.15.2
 pandas>=0.23.1
-scanpy>=1.3.7
 scipy>=1.1.0
 scikit-learn>=0.19.1,!=0.20.0
 tables==3.5.1

